[Docs] Update NNCF QAT docs (#28818)
openvinotoolkit/openvino: docs/articles_en/openvino-workflow/model-optimization-guide/compressing-models-during-training/quantization-aware-training.rst

3. (Optional) Save Checkpoint
##############################

To save a model checkpoint, use the following API:

.. tab-set::

   .. tab-item:: PyTorch
      :sync: pytorch

      .. doxygensnippet:: docs/optimization_guide/nncf/code/qat_torch.py
         :language: python
         :fragment: [save_checkpoint]

   .. tab-item:: TensorFlow 2
      :sync: tensorflow-2

      .. doxygensnippet:: docs/optimization_guide/nncf/code/qat_tf.py
         :language: python
         :fragment: [save_checkpoint]

4. (Optional) Restore from Checkpoint
######################################

To restore the model from checkpoint, use the following API:

.. tab-set::

   .. tab-item:: PyTorch
      :sync: pytorch

      .. doxygensnippet:: docs/optimization_guide/nncf/code/qat_torch.py
         :language: python
         :fragment: [load_checkpoint]

   .. tab-item:: TensorFlow 2
      :sync: tensorflow-2

      .. doxygensnippet:: docs/optimization_guide/nncf/code/qat_tf.py
         :language: python
         :fragment: [load_checkpoint]

Deploying the Quantized Model
##############################

> The model can be converted into the OpenVINO Intermediate Representation (IR) if needed, compiled, and run with OpenVINO without any additional steps.

Reviewer: ```suggestion
You can convert the model to OpenVINO IR, if needed, compile it and run with OpenVINO without
any additional steps.
```
